File: frontend/src/services/wishlistService.js
import axios from "axios";

const WISHLIST_URL = "http://localhost:8087/wishlist";

export const getWishlist = async (userEmail) => {
  const response = await axios.get(WISHLIST_URL, {
    params: {
      userEmail,
    },
  });

  return response.data;
};

export const addToWishlist = async (userEmail, productId) => {
  const response = await axios.post(
    WISHLIST_URL,
    {
      productId,
    },
    {
      params: {
        userEmail,
      },
    }
  );

  return response.data;
};

export const removeFromWishlist = async (wishlistId) => {
  const response = await axios.delete(
    `${WISHLIST_URL}/${wishlistId}`
  );

  return response.data;
};

export const clearWishlist = async (userEmail) => {
  const response = await axios.delete(`${WISHLIST_URL}/clear`, {
    params: {
      userEmail,
    },
  });

  return response.data;
};

export const getWishlistCount = async () => {
  const response = await axios.get(`${WISHLIST_URL}/count`);

  return response.data;
};

File: backend/wishlist-service/src/main/java/com/onedaywear/wishlist/service/WishlistService.java
package com.onedaywear.wishlist.service;

import java.util.List;

import com.onedaywear.wishlist.dto.WishlistRequest;
import com.onedaywear.wishlist.dto.WishlistResponse;

public interface WishlistService {

    WishlistResponse addToWishlist(String userEmail,
                                   WishlistRequest request);

    List<WishlistResponse> getWishlist(String userEmail);

    void removeFromWishlist(Long wishlistId);

    void clearWishlist(String userEmail);

    Long getWishlistCount();
    
}